test_conformance/spirv_new/test_no_integer_wrap_decoration.cpp

```diff
@@ -183,39 +202,76 @@ int test_ext_cl_khr_spirv_no_integer_wrap_decoration(cl_device_id deviceID,
     SPIRV_CHECK_ERROR(err, "Failed to enqueue cl kernel");
 
     std::vector<T> h_res(num);
-    err = clEnqueueReadBuffer(queue, res, CL_TRUE, 0, bytes, &h_res[0], 0, NULL, NULL);
+    err = clEnqueueReadBuffer(queue, res, CL_TRUE, 0, bytes, &h_res[0], 0, NULL,
+                              NULL);
     SPIRV_CHECK_ERROR(err, "Failed to read from ref");
 
-    for (int i = 0; i < num; i++) {
-        if (expected_results[i] != h_res[i]) {
-            log_error("Values do not match at location %d expected = %d got = %d\n", i, expected_results[i], h_res[i]);
+    for (int i = 0; i < num; i++)
+    {
+        if (expected_results[i] != h_res[i])
+        {
+            log_error(
+                "Values do not match at location %d expected = %d got = %d\n",
+                i, expected_results[i], h_res[i]);
             return -1;
         }
     }
 
-    return 0;
+    return TEST_PASS;
 }
 
-#define TEST_FMATH_FUNC(TYPE, FUNC)                                                              \
-    TEST_SPIRV_FUNC(ext_cl_khr_spirv_no_integer_wrap_decoration_##FUNC##_##TYPE)                 \
-    {                                                                                            \
-        return test_ext_cl_khr_spirv_no_integer_wrap_decoration<cl_##TYPE>(deviceID, context, queue, \
-                          "ext_cl_khr_spirv_no_integer_wrap_decoration_"#FUNC"_"#TYPE,           \
-                          #FUNC,                                                                 \
-                          #TYPE                                                                  \
-                          );                                                                     \
+#define TEST_FMATH_FUNC_KHR(TYPE, FUNC)                                        \
+    TEST_SPIRV_FUNC(                                                           \
+        ext_cl_khr_spirv_no_integer_wrap_decoration_##FUNC##_##TYPE)           \
+    {                                                                          \
+        if (!is_extension_available(                                           \
+                deviceID, "cl_khr_spirv_no_integer_wrap_decoration"))          \
+        {                                                                      \
+            log_info("Extension cl_khr_spirv_no_integer_wrap_decoration not "  \
+                     "supported; skipping tests.\n");                          \
+            return TEST_SKIPPED_ITSELF;                                        \
+        }                                                                      \
+        return test_no_integer_wrap_decoration<cl_##TYPE>(                     \
+            deviceID, context, queue,                                          \
+            "ext_cl_khr_spirv_no_integer_wrap_decoration_" #FUNC "_" #TYPE,    \
+            #FUNC, #TYPE);                                                     \
+    }
+
+TEST_FMATH_FUNC_KHR(int, fadd)
```

Comment:
These names are confusing (fadd suggests floating point addition, but this extension only applies to integer types); iadd would make more sense.  However I am aware you're following existing code, so I'm fine if you want to leave the names as they are.

Reply:
Let me know what you'd like to do here.  To fix this we'd need to fix the test names in the .cpp file, and rename all of the .spvasm files.  This isn't too bad and I don't mind making the change, but it could also be somewhat disruptive.

Note, I could just "fix" the SPIR-V 1.4 tests since they're new, but that seems even more confusing, so I'd either like to change the SPIR-V 1.4 tests and the extension tests together, or leave them both as-is.

Reply:
Only renaming the SPIR-V 1.4 tests seems like the worst solution indeed.  I'm inclined to leave the renaming of both as a future PR.

Reply:
Sounds good.  I've filed https://github.com/KhronosGroup/OpenCL-CTS/issues/2057 to ensure we don't forget about this.